src/UhdmAst.cpp

```diff
@@ -54,6 +54,13 @@ void sanitize_str(std::string& s) {
     }
 }
 
+void remove_last_sanitized_index(std::string& s) {
```

Comment:
As opposed to the substitution in `sanitize_str`, this is irreversible. I'd rather avoid making it in-place, return a new string to force the caller to be explicit if it discards the original string.